src/hooks/useHasAppUpdated/useHasAppUpdated.ts

```diff
@@ -3,45 +3,83 @@ import axios from 'axios'
 import isEqual from 'lodash/isEqual'
 import { useMemo, useState } from 'react'
 
-const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
+export const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
 
+const fetchAsset = async (url: string) => {
+  try {
+    // dummy query param bypasses browser cache
+    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
+    return data
+  } catch (e) {
+    console.error(`useHasAppUpdated: error fetching ${url}`, e)
+  }
+}
 export const useHasAppUpdated = () => {
   const [hasUpdated, setHasUpdated] = useState(false)
   const [initialManifestMainJs, setManifestMainJs] = useState(null)
+  const [initialEnv, setEnv] = useState(null)
   // asset-manifest tells us the latest minified built files
-  const url = '/asset-manifest.json'
-  const storeMainManifestJs = async () => {
-    // dummy query param bypasses browser cache
-    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
-    setManifestMainJs(data)
-  }
-  useMemo(storeMainManifestJs, [])
-  useInterval(async () => {
+  const assetManifestUrl = '/asset-manifest.json'
+  const envUrl = '/env.json'
+
+  const storeAndCompareAsset = async () => {
```

Comment:
all async functions must have `try/catch` to avoid an `UnhandledPromiseRejection` error.